common/softwarecontainer-common.cpp

```diff
@@ -93,22 +95,15 @@ bool existsInFileSystem(const std::string &path)
 
 std::string parentPath(const std::string &path_)
 {
-    static constexpr const char *separator = "/";
-    static constexpr const char separator_char = '/';
-
-    auto path = path_;
-
-    // Remove trailing backslashes
-    while ((path.size() > 0) && (path[path.size() - 1] == separator_char)) {
-        path.resize(path.size() - 1);
-    }
+    std::string path2 = path_; // copy constructor
+    char *path = (char *)path2.c_str();
+    std::string parent = dirname(path);
```

Comment:
I actually had it as strdup() before, but wanted to avoid having to use that, since it might fail.